#!/usr/bin/env python
PACKAGE = "nao_walk"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#LIPM
#gen.add("COP_Noise", double_t, 0, "Observer COPX ",    0.35, 0.00,   10.0)
#gen.add("CoM_Noise", double_t, 0, "Observer COPY",    0.5, 0.00,   10.0)
#CT
gen.add("COP_Noise", double_t, 0, "Observer COPX ",    0.35, 0.00,   10.0)
gen.add("CoM_Noise", double_t, 0, "Observer COPY",    0.7, 0.00,   10.0)
gen.add("Observer_CoM",double_t, 0, "Observer CoMX ",    0.4, 0,   10.0)
gen.add("Observer_COP",double_t, 0, "Observer CoMY ",    0.6, 0,   10.0)
gen.add("Kp_Pitch",double_t, 0, "Ankle Pitch P",    0.0, 0,  100.0)
gen.add("Kd_Pitch",double_t, 0, "Ankle Pitch D",    0., 0,   50.0)
gen.add("Kp_Roll",double_t, 0, "Ankle Roll P",    0.0, 0,   100.0)
gen.add("Kd_Roll",double_t, 0, "Ankle Roll D",    0., 0,   50.0)
gen.add("amX",double_t, 0, "Angular Momentum X",    1.0, 0,   3.0)
gen.add("amY",double_t, 0, "Angular Momentum Y",    1.0, 0,   3.0)
gen.add("StepHeight",double_t, 0, "Max Step Height",    0.0165, 0.014,   0.025)
gen.add("kfx",double_t, 0, "ZMP Force Control X",    1, 0,   20.0)
gen.add("kfy",double_t, 0, "ZMP Force Control Y",    2, 0,   20.0)
gen.add("kcx",double_t, 0, "CoM Force Control X",    0.000, 0.0,   15.0)
gen.add("kcy",double_t, 0, "CoM Force Control Y",    0.000, 0.0,   15.0)
exit(gen.generate(PACKAGE, "nao_walk", "GaitControl"))
